### PR TITLE
OWNERS: RelEng and CHANGELOG updates

### PR DESCRIPTION
**Pull request by justaugustus**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

- OWNERS(CHANGELOG): Move reviewers/approvers to CHANGELOG/ dir

  Given we're pulling the relnotes team into review, the changes to
  reviewers will happen a little more frequently, which means we'll have
  the same issue of needing a top-level approver to do updates that we
  were trying to avoid by creating the subdirectory in the first place.

- build/OWNERS: Add Dan and Sascha as reviewers
  
  @hasheddan and @saschagrunert are senior Release Managers, SIG Release leadership, and
  have already been giving thoughtful reviews of the content in this
  directory.

  Adding them as reviewers to spread the load of SIG Release eyes on
  content.

/assign @liggitt @spiffxp @cblecker  
cc: @kubernetes/release-engineering @wilsonehusin 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Comment by liggitt** [comment]
/approve
for moving changelog reviewers to subpackage OWNERS file

/hold in case you want eyes on the build OWNERS change

**Comment by justaugustus** [comment]
Thanks Jordan and Aaron!

/hold cancel
(since Aaron and I are approvers for `build/`)

**Comment by wilsonehusin** [comment]
/lgtm

adding visibility to the 1.21 release notes team as well, this is how Prow finds you and assign you to PRs :smile:

@ashnehete @melodychn @pmmalinov01 @soniasingla

**Pull request merged**
